skeletons/constr_CHOICE_aper.c

```diff
@@ -52,11 +52,14 @@ CHOICE_decode_aper(const asn_codec_ctx_t *opt_codec_ctx,
     } else {
         if(specs->ext_start == -1)
             ASN__DECODE_FAILED;
-        value = aper_get_nsnnwn(pd, ct->upper_bound - ct->lower_bound + 1);
-        if(value < 0) ASN__DECODE_STARVED;
-        value += specs->ext_start;
-        if((unsigned)value >= td->elements_count)
-            ASN__DECODE_FAILED;
+
+        if (ct && ct->upper_bound >= ct->lower_bound) {
+            value = aper_get_nsnnwn(pd, ct->upper_bound - ct->lower_bound + 1);
+            if(value < 0) ASN__DECODE_STARVED;
+            value += specs->ext_start;
+            if((unsigned)value >= td->elements_count)
+                ASN__DECODE_FAILED;
+        }
```

Comment:
What happens if `ct->upped_bound` is less than `ct->lower_bound`? Is it an error condition? 
Are those values always set, i.e., what happens if they are NULL pointers?

Reply:
Well,  neither upper_bound nor lower_bound are pointers. They are `intmax_t` (see declaration in `per_support.h`) so no need to worry about NULL here.
Regarding their values: unfortunately, as I said, I'm no ASN expert. I have no idea what their semantics is, but I'd rather avoid having the code go ahead and calculate `ct->upper_bound - ct->lower_bound + 1` unless upper_bound is actually greater-or-equal than lower_bound. And that's all as far as my reasoning can go here (well, other than fixing the obvious bug where `ct` could be NULL and actually crash here...)

Reply:
Thank you!